server/api/api/endpoints/nuclio.py

```diff
@@ -31,19 +37,103 @@ async def list_api_gateways(
-            .get("labels", {})
-            .get("iguazio.com/username"),
-            _api_gateway.get("metadata", {}).get("name"),
+            project,
+            _api_gateway,
         ),
         auth_info,
     )
+    allowed_api_gateways = {
+        api_gateway: api_gateways[api_gateway] for api_gateway in allowed_api_gateways
+    }
+    return mlrun.common.schemas.APIGatewaysOutput(api_gateways=allowed_api_gateways)
+
+
+@router.get(
+    "/projects/{project}/api-gateways/{gateway}",
+    response_model=mlrun.common.schemas.APIGateway,
+    response_model_exclude_none=True,
+)
+async def get_api_gateway(
+    project: str,
+    gateway: str,
+    auth_info: mlrun.common.schemas.AuthInfo = Depends(deps.authenticate_request),
+):
+    await server.api.utils.auth.verifier.AuthVerifier().query_project_permissions(
+        project_name=project,
+        action=mlrun.common.schemas.AuthorizationAction.read,
+        auth_info=auth_info,
+    )
+    await server.api.utils.auth.verifier.AuthVerifier().query_project_resource_permissions(
+        mlrun.common.schemas.AuthorizationResourceTypes.api_gateway,
+        project,
+        gateway,
+        mlrun.common.schemas.AuthorizationAction.read,
+        auth_info,
+    )
+    client = server.api.utils.clients.async_nuclio.Client(auth_info)
+    try:
+        api_gateway = await client.get_api_gateway(project_name=project, name=gateway)
+    finally:
+        await client.close_session()
+    return api_gateway
+
+
+@router.put(
+    "/projects/{project}/api-gateways/{gateway}",
+    response_model=mlrun.common.schemas.APIGateway,
+    response_model_exclude_none=True,
+)
+async def store_api_gateway(
+    project: str,
+    gateway: str,
+    api_gateway: mlrun.common.schemas.APIGateway,
+    auth_info: mlrun.common.schemas.AuthInfo = Depends(deps.authenticate_request),
+):
+    await server.api.utils.auth.verifier.AuthVerifier().query_project_permissions(
+        project_name=project,
+        action=mlrun.common.schemas.AuthorizationAction.read,
+        auth_info=auth_info,
+    )
+    await server.api.utils.auth.verifier.AuthVerifier().query_project_resource_permissions(
+        mlrun.common.schemas.AuthorizationResourceTypes.api_gateway,
+        project,
+        gateway,
+        mlrun.common.schemas.AuthorizationAction.store,
+        auth_info,
+    )
+    client = server.api.utils.clients.async_nuclio.Client(auth_info)
+    try:
+        if not await client.api_gateway_exists(
+            name=gateway,
+            project_name=project,
+        ):
+            await client.store_api_gateway(
+                project_name=project,
+                api_gateway_name=gateway,
+                api_gateway=api_gateway,
+                create=True,
+            )
+        else:
+            await client.store_api_gateway(
+                project_name=project,
+                api_gateway_name=gateway,
+                api_gateway=api_gateway,
+            )
```

Comment:
```suggestion
        create = not await client.api_gateway_exists(
            name=gateway,
            project_name=project,
        )
        await client.store_api_gateway(
                project_name=project,
                api_gateway_name=gateway,
                api_gateway=api_gateway,
                create=create,
            )
```

i think shorter and concise